### PR TITLE
fix ide scripts; users can run ide.sh from any cwd

### PR DESCRIPTION
**Pull request by qinkunbao**
## Description
fix ide scripts; users can run ide.sh from any cwd

## Type of change (select or add applied and delete the others)

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] API change with a documentation update
- [ ] Additional test coverage
- [ ] Code cleanup or just sync with upstream third-party crates

## Checklist

- [ ] Fork the repo and create your branch from `master`.
- [ ] If you've added code that should be tested, add tests.
- [ ] If you've changed APIs, update the documentation.
- [ ] Ensure the tests pass (see CI results).
- [ ] Make sure your code lints/format.

### CODE REVIEW

File: scripts/ide.sh [hunk 1]
```diff
@@ -17,20 +17,21 @@
 # specific language governing permissions and limitations
 # under the License.
 
-TRUSTED_TOML="cmake/tomls/Cargo.sgx_trusted_lib.toml"
-UNTRUSTED_TOML="cmake/tomls/Cargo.sgx_untrusted_app.toml"
+script_dir=$(cd "$(dirname "${BASH_SOURCE[0]}")" &>/dev/null && pwd -P)
```
Comment (mssun) [inline scripts/ide.sh:20]: Can we use ALL_CAPS for these variables?

**Review by ya0guang — COMMENTED** [review]
Thanks for the changes. I also believe using CAPS is a good idea to represent vars.

File: scripts/ide.sh (hunk 1 above)
Comment (dingelish) [reply scripts/ide.sh:20]: https://stackoverflow.com/questions/673055/correct-bash-and-shell-script-variable-capitalization @qinkunbao please read and see if capitalization is needed

Comment (mssun) [reply scripts/ide.sh:20]: Do we have a shell style guides for reference? I found one: https://google.github.io/styleguide/shellguide.html. If it is worth to follow, we can put it into our docs.

Comment (qinkunbao) [reply scripts/ide.sh:20]: > https://stackoverflow.com/questions/673055/correct-bash-and-shell-script-variable-capitalization @qinkunbao please read and see if capitalization is needed

Thanks for the information. It looks like variables like `TRUSTED_TOML` and `UNTRUSTED_TOML` in this shell script should be the lower case as well, according to the convention.

Comment (qinkunbao) [reply scripts/ide.sh:20]: Some bash scripts under the `cmake/scripts` folder do not the following the convention.

For example,

https://github.com/apache/incubator-teaclave/blob/c09805e998174034e491e1b46258b7308dc0789d/cmake/scripts/sgx_link_sign.sh#L35-L38
https://github.com/apache/incubator-teaclave/blob/c09805e998174034e491e1b46258b7308dc0789d/cmake/scripts/gen_enclave_sig.sh#L30

Comment (mssun) [reply scripts/ide.sh:20]: > Some bash scripts under the `cmake/scripts` folder do not the following the convention.
> 
> For example,
> 
> https://github.com/apache/incubator-teaclave/blob/c09805e998174034e491e1b46258b7308dc0789d/cmake/scripts/sgx_link_sign.sh#L35-L38
> 
> 
> https://github.com/apache/incubator-teaclave/blob/c09805e998174034e491e1b46258b7308dc0789d/cmake/scripts/gen_enclave_sig.sh#L30

Don't worry. We can improve this gradually. You can crate a new issue to track this legacy code style issue.

**Review by mssun — APPROVED** [review]

File: scripts/ide.sh [hunk 2]
```diff
@@ -17,50 +17,51 @@
...
 copy() {
     # $1: TOML
     # $2: LOCK
     # $3: CONFIG
-    mkdir ${teaclave_root}/.cargo
-    cp $1 ${teaclave_root}/${TOML_DEST}
-    cp $2 ${teaclave_root}/${LOCK_DEST}
-    cp $3 ${teaclave_root}/${CONFIG_DEST}
+    mkdir "${teaclave_root}"/.cargo
+    cp "$1" "${teaclave_root}"/${toml_dest}
```
Comment (bigdata-memory) [inline scripts/ide.sh:41]: I could be more readable if assign a new variable name for this first argument.

**Review by mssun — APPROVED** [review]

**Pull request merged**